src/guiguts/tools/jeebies.py

```diff
@@ -46,193 +64,580 @@ def check_for_jeebies_in_file(self) -> None:
+                    # We have the start position in the paragraph string of its hebe.
+                    # Locate the corresponding file line and its column position on
+                    # that line.
 
-            rlim = min(rlim, len(para))
+                    line_number, hebe_position_on_line = self.locate_file_line(
+                        paragraph_number,
+                        hebe_start,
+                        paragraph_start_line_numbers,
+                        paragraph_line_boundaries,
+                    )
+                    line = file_lines_list[line_number - 1]
+
+                    self.add_to_dialog(
+                        info,
+                        line,
+                        line_number,
+                        hebe_position_on_line,
+                        checker_dialog,
+                    )
 
-        elif indx == len(para) - 1:
-            # Required slice is the tail of the paragraph string.
-            rlim = len(para)
-            llim = rlim - 60
+            return count_of_suspects
 
-            llim = max(llim, 0)
+        ####
+        # Find and report 2-word hebe suspects in paragraph text
+        ####
+
+        suspects_count = 0
+        # Get the opposite word to the one being reported on.
+        behe = "he" if hebe == "be" else "be"
+
+        # Look for 2-form hebe phrases. There are two types:
+        #   1. The hebe is the first word of the phrase: e.g. '“Be diligent ....”'
+        #   2. The hebe is the second word of the phrase: e.g. '“... could be.”'
+        # Note that the hebe in a 2-form phrase is ALWAYS delimited by at least
+        # one punctuation character, as in the examples above, or a punctuation
+        # character followed by a space as in '... had done. He then ...'. Special
+        # treatment needed when phrase involves a contaction as in 'an’ be damned.'
+        # See below for this.
+
+        if order == "first":
+            # Looking for hebe as first word of 2-word punctuation delimited phrase.
+
+            para_lc = paragraph_text.lower()
+
+            # NB A hebe phrase containing a contraction as in 'an’ be damned' has already
+            #    been processed as a 3-form phrase. Avoid it being processed again as a
+            #    2-form phrase. Obscure the right curly quote characters whenever they
+            #    and a space precede a hebe.
+            qs_hebe = r"’ " + f"{hebe}"
+            ss_hebe = r"  " + f"{hebe}"
+            para_lc = re.sub(qs_hebe, ss_hebe, para_lc)
+            # Look for type 1, 2-form hebe phrases in the edited paragraph string.
+            regx = r"(?<=^|\p{P}|\p{P} )" + f"{hebe}" + " [a-z]+"
+            suspects_count += do_finditer_and_report()
+
+        elif order == "second":
+            # Looking for hebe as second word of 2-word punctuation delimited phrase.
+
+            para_lc = paragraph_text.lower()
+
+            regx = "[a-z]+ " + f"{hebe}" + r"(?=\p{P}|$)"
```

Comment:
I would merge all of this into a single f-string

Reply:
Done. And similarly elsewhere. I used concatenation there because at the time I had not discovered how to escape literal '{' and '}' in an f-string.